File: frontend/bff/src/application/files/queries/list-file-lines.query.ts
import type { FileLineStatus } from '../../../cross-cutting/enums/file-line-status.js';
import type { PageRequest } from '../../../cross-cutting/paging/page.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface ListFileLinesQuery {
  context: RequestContext;
  fileId: string;
  /** null = todas. */
  status: FileLineStatus | null;
  page: PageRequest;
}

File: backend/storage-service/src/Fix.Storage.Application/UseCases/Files/Queries/ListFileLinesQuery.cs
using Fix.Storage.Application.Abstractions.Authorization;
using Fix.Storage.Application.Abstractions.Context;
using Fix.Storage.Application.Abstractions.Messaging;
using Fix.Storage.Application.Files.Dtos;
using Fix.Storage.Domain.AggregateRoots.Files;

namespace Fix.Storage.Application.Files.Queries;

[RequireMembership]
public sealed record ListFileLinesQuery(Guid UserId, Guid OrganizationId, Guid FileId, FileLineStatus? Status, int Page, int PageSize)
    : IQuery<PagedList<FileLineDto>>, IOrganizationRequest;
